File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    function inorderTraversal($root) {
        $r = array();
        $this->traverse($root, $r);
        return $r;
    }
    
    function traverse($n, &$x)
    {
        if($n->left != null)
        {
            $this->traverse($n->left, $x);
        }
        array_push($x, $n->val);
        if($n->right != null)
        {
            $this->traverse($n->right, $x);
        }
    }
} 

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun inorderTraversal(root: TreeNode?): List<Int> {
        if(root == null)
            return mutableListOf();
        val s  = mutableListOf<Int>();
        for(a in inorderTraversal(root.left))
            s.add(a);
        s.add(root.`val`);
        for(a in inorderTraversal(root.right))
            s.add(a);
        return s;
    }
}